Selfie

Selfie-17 opened the pull request:
Fixed the previous PR issues #25.

Selfie-17 committed:
Added dark and light mode toggle to it

Selfie-17 committed:
Guest mode removes. and modified the changes told

Selfie-17 committed:
Merge branch 'main' into selfie

Selfie-17 committed:
Fixed the darkmode issues

Selfie-17 commented:
@Swastik007sharma Please verify it now

Swastik007sharma commented on frontend/src/components/ItemCard.jsx:
```diff
@@ -18,8 +18,8 @@ const ItemCard = ({
   setOtp,           // Function to update OTP
 }) => {
   return (
-    <div className="bg-white border border-gray-200 rounded-xl shadow-sm hover:shadow-xl transition-all duration-300 ease-in-out overflow-hidden w-full relative transform hover:scale-[1.02] group">
```
You should keep these transitions

Swastik007sharma commented on frontend/src/components/ItemCard.jsx:
```diff
@@ -18,8 +18,8 @@ const ItemCard = ({
 }) => {
   return (
-    <div className="bg-white border border-gray-200 rounded-xl shadow-sm hover:shadow-xl transition-all duration-300 ease-in-out overflow-hidden w-full relative transform hover:scale-[1.02] group">
-      <div className="relative w-full h-48 sm:h-56 md:h-64 bg-gray-100 rounded-t-xl overflow-hidden">
```
this as well

Selfie-17 commented:
@Swastik007sharma can you tell me in brief. About them

In brief, you should keep the previous transitions and shadow on hover mainly in ItemDetail.jsx and ItemCard.jsx. Also, keep the placeholder for Images as they were

Swastik007sharma commented on frontend/src/components/ItemCard.jsx:
```diff
@@ -28,35 +28,22 @@ const ItemCard = ({
             onClick={() => window.open(item.image, '_blank')}
           />
         ) : (
-          <div className="w-full h-full flex flex-col items-center justify-center bg-gradient-to-br from-gray-50 to-gray-100 group-hover:from-blue-50 group-hover:to-indigo-50 transition-all duration-300 ease-in-out">
```
Keep these changes as well, and they look more modern. from line no. 31 to 45, only add variables as required,

Swastik007sharma commented on frontend/src/components/ItemCard.jsx:
```diff
@@ -28,35 +28,22 @@ const ItemCard = ({
             <button
               onClick={onDelete}
-              className="text-red-600 hover:text-red-800 p-2 rounded-full bg-white bg-opacity-80 hover:bg-opacity-100 transition-all duration-200 hover:scale-110 shadow-sm hover:shadow-md"
+              className="bg-red-500 hover:bg-red-600 text-white p-2 rounded-full transition-colors duration-200"
```
Add shadow and opacity on hover

Swastik007sharma commented on frontend/src/components/ItemCard.jsx:
```diff
@@ -28,35 +28,22 @@ const ItemCard = ({
             <button
               onClick={onEdit}
-              className="text-blue-600 hover:text-blue-800 p-2 rounded-full bg-white bg-opacity-80 hover:bg-opacity-100 transition-all duration-200 hover:scale-110 shadow-sm hover:shadow-md"
+              className="bg-blue-500 hover:bg-blue-600 text-white p-2 rounded-full transition-colors duration-200"
```
Add shadow and opacity on hover

Selfie-17 committed:
Fixed the otp and transition effect

Selfie-17 committed:
fixed

Selfie-17 commented:
@Swastik007sharma I pushed the code but this time i am unable to create a pull request.

https://github.com/Selfie-17/CampusTrack/tree/selfie       This is my github code. 

Please check it and tell me how to raise a pr

Selfie-17 committed:
fixed

Swastik007sharma commented on frontend/src/pages/ItemDetails.jsx:
```diff
@@ -370,78 +371,68 @@ function ItemDetails() {
-                        </p>
-                      </div>
-                    </div>
+                  <div className="w-full h-64 rounded-lg bg-gray-200 dark:bg-gray-700 flex items-center justify-center">
```
Keep it as it was before the change from line 373 to 287 before the changes

Replace the below code with the code in line no. 374 & 375
``` javascript
<div className="w-full h-64 bg-gradient-to-br from-gray-50 to-gray-100 rounded-lg flex items-center justify-center group hover:from-blue-50 hover:to-indigo-50 transition-all duration-300 ease-in-out">
     <div className="flex flex-col items-center space-y-3 p-6 rounded-lg bg-white bg-opacity-80 backdrop-blur-sm shadow-sm group-hover:shadow-md transition-all duration-300">
          <div className="relative">
              <FaSearch className="text-gray-400 text-4xl group-hover:text-blue-400 transition-colors duration-300" />
              <FaImage className="text-gray-300 text-2xl absolute -bottom-1 -right-1 group-hover:text-blue-300 transition-colors duration-300" />
          </div>
              <div className="text-center">
              <p className="text-gray-500 text-sm font-medium group-hover:text-gray-600 transition-colors duration-300">
                   No Image Available
              </p>
            </div>
      </div>
```

Swastik007sharma reviewed (changes requested):
Make these changes, and the ones where I left comments as well

Selfie-17 commented:
> Make these changes, and the ones where I left comments as well

Can you send the screenshots

Swastik007sharma commented:
> > Make these changes, and the ones where I left comments as well
> 
> Can you send the screenshots

previous Placeholder for image 
<img width="278" height="256" alt="image" src="https://github.com/user-attachments/assets/034223e3-a495-49bc-ae84-fc9c8fbe1aa7" />

Current placeholder for image, which needs to be changed
<img width="347" height="280" alt="image" src="https://github.com/user-attachments/assets/ae332811-e244-4b9a-adf1-93b79183f685" />

Selfie-17 committed:
Fixed no image available image and made it adaptable to both dark adn light mode

Selfie-17 commented:
> > > Make these changes, and the ones where I left comments as well
> > 
> > 
> > Can you send the screenshots
> 
> previous Placeholder for image <img alt="image" width="278" height="256" src="https://private-user-images.githubusercontent.com/111640999/473771674-034223e3-a495-49bc-ae84-fc9c8fbe1aa7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQyMzc2NTMsIm5iZiI6MTc1NDIzNzM1MywicGF0aCI6Ii8xMTE2NDA5OTkvNDczNzcxNjc0LTAzNDIyM2UzLWE0OTUtNDliYy1hZTg0LWZjOWM4ZmJlMWFhNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgwM1QxNjA5MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZDExOThiNjIxNTZmOThmZjc2MzdkMWJkYzA4YzkzZDIwZGE1YzJmODlmODRlMWIwZGEwNDljOWU2ZTE4NzkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HCtsc8FIOXEvm7JEdhOcRIxWL49fehCZgpvYcGLvTLY">
> 
> Current placeholder for image, which needs to be changed <img alt="image" width="347" height="280" src="https://private-user-images.githubusercontent.com/111640999/473771735-ae332811-e244-4b9a-adf1-93b79183f685.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQyMzc2NTMsIm5iZiI6MTc1NDIzNzM1MywicGF0aCI6Ii8xMTE2NDA5OTkvNDczNzcxNzM1LWFlMzMyODExLWUyNDQtNGI5YS1hZGYxLTkzYjc5MTgzZjY4NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgwM1QxNjA5MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02Mjk5M2ZjNWI0MTllMTU2MGYzZmQyMTU0Y2NjYjU4MThlOWM3ZGZkMGJkNjAxOTI4YjJjOTJkYzE3MzlhMjY3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5sP0rcCizpN5pFf62tib_u-KOYFoJNol4EKyXNCWlEs">

I have fixed this issue. Those animations you have told I havent understood . Please consider the work till now. The other conflicts we will do in a new issue. Try to mail your whatsapp number to my mail kampadevaselvaraj@gmail.com @Swastik007sharma . I have fixed most of them

Selfie-17 committed:
fixed the transitions
